packages/Interface/src/DTK_C_API.cpp

```diff
@@ -0,0 +1,434 @@
+{
+    auto u = get_function<DTK_node_list_data_function>( user_data );
+    u.first( u.second, coordinates.data(), is_ghost_node.data() );
+}
+
+void BoundingVolumeListSizeFunctionWrapper( std::shared_ptr<void> user_data,
+                                            unsigned &space_dim,
+                                            size_t &local_num_volumes,
+                                            bool &has_ghosts )
+{
+    auto u = get_function<DTK_bounding_volume_list_size_function>( user_data );
+    u.first( u.second, &space_dim, &local_num_volumes, &has_ghosts );
+}
+
+void BoundingVolumeListDataFunctionWrapper( std::shared_ptr<void> user_data,
+                                            View<Coordinate> bounding_volumes,
+                                            View<bool> is_ghost_volume )
+{
+    auto u = get_function<DTK_bounding_volume_list_data_function>( user_data );
+    u.first( u.second, bounding_volumes.data(), is_ghost_volume.data() );
+}
+
+void PolyhedronListSizeFunctionWrapper(
+    std::shared_ptr<void> user_data, unsigned &space_dim,
+    size_t &local_num_nodes, size_t &local_num_faces,
+    size_t &total_nodes_per_face, size_t &local_num_cells,
+    size_t &total_faces_per_cell, bool &has_ghosts )
+{
+    auto u = get_function<DTK_polyhedron_list_size_function>( user_data );
+    u.first( u.second, &space_dim, &local_num_nodes, &local_num_faces,
+             &total_nodes_per_face, &local_num_cells, &total_faces_per_cell,
+             &has_ghosts );
+}
+
+void PolyhedronListDataFunctionWrapper(
+    std::shared_ptr<void> user_data, View<Coordinate> coordinates,
+    View<LocalOrdinal> faces, View<unsigned> nodes_per_face,
+    View<LocalOrdinal> cells, View<unsigned> faces_per_cell,
+    View<int> face_orientation, View<bool> is_ghost_cell )
+{
+    auto u = get_function<DTK_polyhedron_list_data_function>( user_data );
+    u.first( u.second, coordinates.data(), faces.data(), nodes_per_face.data(),
+             cells.data(), faces_per_cell.data(), face_orientation.data(),
+             is_ghost_cell.data() );
+}
+
+void CellListSizeFunctioWrapper( std::shared_ptr<void> user_data,
+                                 unsigned &space_dim, size_t &local_num_nodes,
+                                 size_t &local_num_cells,
+                                 unsigned &nodes_per_cell, bool &has_ghosts )
+{
+    auto u = get_function<DTK_cell_list_size_function>( user_data );
+    u.first( u.second, &space_dim, &local_num_nodes, &local_num_cells,
+             &nodes_per_cell, &has_ghosts );
+}
+
+void CellListDataFunctionWrapper( std::shared_ptr<void> user_data,
+                                  View<Coordinate> coordinates,
+                                  View<LocalOrdinal> cells,
+                                  View<bool> is_ghost_cell,
+                                  std::string &cell_topology )
+{
+    const int max_string_size = 255;
+    std::vector<char> c_cell_topology( max_string_size );
```

Comment:
How about you make `max_string_size` a `constexpr` and use `char[max_string_size]`?